samples/image-uploader/src/index.js

```diff
@@ -0,0 +1,437 @@
+import React from "react"
+import PropTypes from "prop-types"
+import ReactDOM from "react-dom"
+import "@contentful/forma-36-react-components/dist/styles.css"
+
+import { Spinner } from "@contentful/forma-36-react-components"
+import { init } from "contentful-ui-extensions-sdk"
+import UploadView from "./components/UploadView"
+import ProgressView from "./components/ProgressView"
+import FileView from "./components/FileView"
+import {
+  readFileAsUrl,
+  getImageUrlFromDataTransfer,
+  getAssetIdFromDataTransfer,
+  getBase64FromDataTransfer,
+  getMimeTypeByPath
+} from "./utils"
+
+import "./index.css"
+
+class App extends React.Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired
+  }
+
+  state = {
+    isDraggingOver: false,
+    value: this.props.sdk.field.getValue(this.findProperLocale())
+  }
+
+  componentDidMount() {
+    this.props.sdk.window.startAutoResizer()
+
+    // Handler for external field value changes (e.g. when multiple authors are working on the same entry).
+    this.detachExternalChangeHandler = this.props.sdk.field.onValueChanged(
+      this.onExternalChange
+    )
+
+    if (this.state.value) {
+      this.props.sdk.space
+        .getAsset(this.state.value.sys.id)
+        .then(asset => this.setState({ asset }))
+        .catch(this.onError)
```

Comment:
How do you feel about improving the error reporting here. As `value.sys.id` might reference a deleted asset `getAsset` might fail not only because of network errors. In case of a deleted asset we could either

a) clear the current value in the field and show the extension as if the field was empty in the first place (this might lead to problems if there is a validation on the field)
b) show the user an appropriate error message that indicates that he needs to remove the current value and upload or reference another asset.

In any case we should allow the user to make changes to the value of the field. However the current experience just shows an endless spinner and a very generic "request failed error"

![Screenshot 2019-03-20 at 15 52 27](https://user-images.githubusercontent.com/84460/54694618-c527b680-4b28-11e9-9f3e-202d73dd2169.png)

Reply:
@hwartig good point, fixed